### PR TITLE
Add tests for Thread constructor with stack size argument

### PR DESCRIPTION
**Pull request by kouvel**
Fixes dotnet/coreclr#9158
Fixes dotnet/coreclr#9167

Part of https://github.com/dotnet/coreclr/pull/9186
Forked from https://github.com/dotnet/corefx/pull/15576

### CODE REVIEW

**Comment by kouvel** [comment]
@janvorli @jkotas

File: src/System.Threading.Thread/tests/ThreadTests.netstandard.cs [hunk 1]
```diff
@@ -20,10 +20,21 @@ public static class ThreadTests
...
+                t =>
+                {
+                    t.IsBackground = true;
+                    t.Start();
+                    Assert.True(t.Join(UnexpectedTimeoutMilliseconds));
+                };
+            startThreadAndJoin(new Thread(() => { }));
+            startThreadAndJoin(new Thread(() => { }, 0));
+            startThreadAndJoin(new Thread(() => { }, 64 << 10)); // 64 KB
+            startThreadAndJoin(new Thread(() => { }, 16 << 20)); // 16 MB
```
Comment (jkotas) [inline src/System.Threading.Thread/tests/ThreadTests.netstandard.cs]: Would it make sense to verify that the stack is really 16MB by doing stackalloc for close to 16MB of stuff?

Comment (kouvel) [inline src/System.Threading.Thread/tests/ThreadTests.netstandard.cs]: Done

File: src/System.Threading.Thread/tests/ThreadTests.netstandard.cs [hunk 2]
```diff
@@ -20,10 +20,34 @@ public static class ThreadTests
...
+                stackSizeBytes =>
+                {
+                    // Try to stack-allocate an array to verify that close to the expected amount of stack space is actually
+                    // available
+                    int bufferSizeBytes = Math.Max(16 << 10, stackSizeBytes - (64 << 10));
+                    unsafe
+                    {
+                        byte* buffer = stackalloc byte[bufferSizeBytes];
+                        buffer[0] = 0xff;
+                        buffer[bufferSizeBytes - 1] = 0xff;
```
Comment (jkotas) [inline src/System.Threading.Thread/tests/ThreadTests.netstandard.cs]: Maybe better to use Volatile.Write to make sure that it won't get optimized out.

Comment (kouvel) [inline src/System.Threading.Thread/tests/ThreadTests.netstandard.cs]: Done

**Comment by stephentoub** [comment]
Thanks, @kouvel.

**Pull request merged**